packages/amplify-graphql-api-construct-tests/src/__tests__/owner-auth/bind-sql-ids/bind-sql-ids.test.ts

```diff
@@ -0,0 +1,252 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+
+import { createNewProjectDir, deleteProjectDir } from 'amplify-category-api-e2e-core';
+import * as generator from 'generate-password';
+import { DDB_AMPLIFY_MANAGED_DATASOURCE_STRATEGY } from '@aws-amplify/graphql-transformer-core';
+import { initCDKProject, cdkDeploy, cdkDestroy } from '../../../commands';
+import {
+  createCognitoUser,
+  dbDetailsToModelDataSourceStrategy,
+  signInCognitoUser,
+  TestDefinition,
+  writeStackPrefix,
+  writeTestDefinitions,
+} from '../../../utils';
+import { SqlDatabaseDetails, SqlDatatabaseController } from '../../../sql-datatabase-controller';
+import { testProtectsHasMany, testProtectsHasOne } from './test-implementations';
+
+jest.setTimeout(1000 * 60 * 60 /* 1 hour */);
+
+/**
+ * We have separate tests to ensure we properly escape/bind incoming SQL variables. These E2Es ensure that binding carries over to
+ * reference fields in a relationship, since the IDs & primary keys have special handling in the resolvers.
+ */
+describe('Reference IDs pointing to SQL data sources are bound to prevent SQL injection', () => {
```

Comment:
You'll note that this suite only contains test combinations that have SQL-based primary models. We're testing to ensure that reference fields with SQL-injection content can't subvert auth rules, and the reference field always points from Related -> Primary.